[Help][Travis] Cypress failures in CI

- [x] Fails to search for Institution (Element detached from DOM)
- [x] Institutions API fails to start https://github.com/cfpb/hmda-platform/issues/3996
- [ ] Institutions API unable to read/write from History table (Moved to ticket #906 for deeper investigation)
- [x] Re-enable Travis builds for branches and PRs once the API is updated.
- [x] Fix HMDA Help stuck-in-auth-loop (#884)
- [x] Update Change Log specs to match recent additions (#884)
- [x] Remove Data Browser specs from non-Prod environments
- [x] Use Prod api for /rateSpread outside of CI environments

Seeing a new error on Travis, which may be causing the cause of `HMDA Help` tests failing.

I have deleted the TravisCI caches to enable a fresh build (similar to what I had to do on my local machine) and have triggered a re-run on [here](https://travis-ci.org/github/cfpb/hmda-frontend/builds/756777078) to test if this resolves the issue. 

Log snippet of the error, which occurs when creating the Institution test data:
```
yarn run v1.22.4
$ yarn newman run cypress/ci/tests/CREATE_INSTITUTIONS.postman_collection.json -e cypress/ci/config/PLATFORM_ENV.postman_environment.json -d cypress/ci/config/institutions.json
$ /home/travis/build/cfpb/hmda-frontend/node_modules/.bin/newman run cypress/ci/tests/CREATE_INSTITUTIONS.postman_collection.json -e cypress/ci/config/PLATFORM_ENV.postman_environment.json -d cypress/ci/config/institutions.json
newman
CI HMDA Filing Setup
Iteration 1/12
→ Filing API Health Check
  ┌
  │ undefined
  └
  GET localhost:8080 [2021-01-30 00:54:58,561][INFO ][hmda2-akka.actor.default-dispatcher-16] [200] GET http://localhost:8080/ took: 17 ms
[2021-01-30 00:54:58,562][INFO ][hmda2-akka.actor.default-dispatcher-16] [200] GET http://localhost:8080/ took: 20 ms
[200 OK, 303B, 108ms]
  ✓  Status code is 200
  ✓  Filing API Health Check
→ Create (PUT) Institution - 8081
  ┌
  │ undefined
  └
  PUT localhost:8081/institutions [2021-01-30 00:54:59,062][INFO ][hmda2-akka.actor.default-dispatcher-6] Started Institution: Institution-FRONTENDTESTBANK9999
[2021-01-30 00:54:59,337][INFO ][hmda2-cassandra-plugin-default-dispatcher-18] Detected Guava >= 19 in the classpath, using modern compatibility layer
[2021-01-30 00:54:59,442][INFO ][hmda2-cassandra-plugin-default-dispatcher-19] DataStax Java driver 3.7.2 for Apache Cassandra
[2021-01-30 00:54:59,845][INFO ][hmda2-cassandra-plugin-default-dispatcher-19] Using native clock to generate timestamps.
[2021-01-30 00:54:59,845][INFO ][hmda2-cassandra-plugin-default-dispatcher-20] Using native clock to generate timestamps.
[2021-01-30 00:55:00,063][INFO ][hmda2-cassandra-plugin-default-dispatcher-20] Cassandra slow queries enabled for hmda2_journal with threshold 1000
[2021-01-30 00:55:00,068][INFO ][hmda2-cassandra-plugin-default-dispatcher-19] Cassandra slow queries enabled for hmda2_snapshot with threshold 1000
[2021-01-30 00:55:00,251][INFO ][hmda2-cassandra-plugin-default-dispatcher-17] PID obtained through native call to getpid(): 6047
[2021-01-30 00:55:00,285][INFO ][hmda2-cassandra-plugin-default-dispatcher-18] Using native clock to generate timestamps.
[2021-01-30 00:55:00,285][INFO ][hmda2-cassandra-plugin-default-dispatcher-18] Cassandra slow queries enabled for hmda2_journal with threshold 1000
[2021-01-30 00:55:00,360][WARN ][hmda2-cassandra-plugin-default-dispatcher-18] Found Netty's native epoll transport in the classpath, but epoll is not available. Using NIO instead.
java.lang.UnsatisfiedLinkError: could not load a native library: netty_transport_native_epoll_x86_64
	at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:224)
	at io.netty.channel.epoll.Native.loadNativeLibrary(Native.java:231)
	at io.netty.channel.epoll.Native.<clinit>(Native.java:58)
	at io.netty.channel.epoll.Epoll.<clinit>(Epoll.java:39)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:315)
	at com.datastax.driver.core.NettyUtil.<clinit>(NettyUtil.java:57)
	at com.datastax.driver.core.NettyOptions.eventLoopGroup(NettyOptions.java:95)
	at com.datastax.driver.core.Connection$Factory.<init>(Connection.java:933)
	at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1499)
	at com.datastax.driver.core.Cluster.init(Cluster.java:208)
	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:376)
	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:355)
	at akka.persistence.cassandra.ConfigSessionProvider.$anonfun$connect$1(ConfigSessionProvider.scala:50)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:56)
	at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:93)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:93)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:48)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:48)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
	Suppressed: java.lang.UnsatisfiedLinkError: could not load a native library: netty_transport_native_epoll
		at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:224)
		at io.netty.channel.epoll.Native.loadNativeLibrary(Native.java:234)
		... 27 common frames omitted
	Caused by: java.io.FileNotFoundException: META-INF/native/libnetty_transport_native_epoll.so
		at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:173)
		... 28 common frames omitted
		Suppressed: java.lang.UnsatisfiedLinkError: no netty_transport_native_epoll in java.library.path: [/usr/java/packages/lib, /usr/lib64, /lib64, /lib, /usr/lib]
			at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2660)
			at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:829)
			at java.base/java.lang.System.loadLibrary(System.java:1867)
			at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
			at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:349)
			at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:136)
			... 28 common frames omitted
			Suppressed: java.lang.UnsatisfiedLinkError: no netty_transport_native_epoll in java.library.path: [/usr/java/packages/lib, /usr/lib64, /lib64, /lib, /usr/lib]
				at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2660)
				at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:829)
				at java.base/java.lang.System.loadLibrary(System.java:1867)
				at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
				at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
				at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
				at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
				at java.base/java.lang.reflect.Method.invoke(Method.java:566)
				at io.netty.util.internal.NativeLibraryLoader$1.run(NativeLibraryLoader.java:369)
				at java.base/java.security.AccessController.doPrivileged(Native Method)
				at io.netty.util.internal.NativeLibraryLoader.loadLibraryByHelper(NativeLibraryLoader.java:361)
				at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:339)
				... 29 common frames omitted
Caused by: java.io.FileNotFoundException: META-INF/native/libnetty_transport_native_epoll_x86_64.so
	at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:173)
	... 28 common frames omitted
	Suppressed: java.lang.UnsatisfiedLinkError: no netty_transport_native_epoll_x86_64 in java.library.path: [/usr/java/packages/lib, /usr/lib64, /lib64, /lib, /usr/lib]
		at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2660)
		at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:829)
		at java.base/java.lang.System.loadLibrary(System.java:1867)
		at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
		at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:349)
		at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:136)
		... 28 common frames omitted
		Suppressed: java.lang.UnsatisfiedLinkError: no netty_transport_native_epoll_x86_64 in java.library.path: [/usr/java/packages/lib, /usr/lib64, /lib64, /lib, /usr/lib]
			at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2660)
			at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:829)
			at java.base/java.lang.System.loadLibrary(System.java:1867)
			at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
			at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
			at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
			at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
			at java.base/java.lang.reflect.Method.invoke(Method.java:566)
			at io.netty.util.internal.NativeLibraryLoader$1.run(NativeLibraryLoader.java:369)
			at java.base/java.security.AccessController.doPrivileged(Native Method)
			at io.netty.util.internal.NativeLibraryLoader.loadLibraryByHelper(NativeLibraryLoader.java:361)
			at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:339)
			... 29 common frames omitted
```

--- a/cypress/integration/data-browser/2018.spec.js
+++ b/cypress/integration/data-browser/2018.spec.js
@@ -4,8 +4,8 @@ const { HOST, ENVIRONMENT } = Cypress.env()
 const dbUrl = dbURL.bind(null, HOST)
 
 describe('Data Browser 2018', function () {
-  if(isCI(ENVIRONMENT)) it("Does not run on CI")
-  else if((!isBeta(HOST) || (isBeta(HOST) && !isProd(HOST)))){
+  if(!isProd(HOST)) it("Only runs in Production")
+  else {
     it('State/Institution/PropertyType', function () {
       cy.get({ HOST, ENVIRONMENT }).logEnv()
       cy.viewport(1000, 940)
@@ -171,7 +171,5 @@ describe('Data Browser 2018', function () {
         })
       }
     })
-  } else {
-    it(`does not run on host: ${HOST}`)
   }
 })

--- a/cypress/integration/data-browser/2019.spec.js
+++ b/cypress/integration/data-browser/2019.spec.js
@@ -4,8 +4,8 @@ const { HOST, ENVIRONMENT } = Cypress.env()
 const dbUrl = dbURL.bind(null, HOST)
 
 describe('Data Browser 2019', function () {
-  if(isCI(ENVIRONMENT)) it("Does not run on CI")
-  else if(!isBeta(HOST)){
+  if(!isProd(HOST)) it("Only runs in Production")
+  else {
     it('State/Institution/PropertyType', function () {
       cy.get({ HOST, ENVIRONMENT }).logEnv()
       cy.viewport(1000, 940)
@@ -172,7 +172,5 @@ describe('Data Browser 2019', function () {
         })
       }
     })
-  } else {
-    it(`does not run on host: ${HOST}`)
   }
 })

--- a/src/common/environmentChecks.js
+++ b/src/common/environmentChecks.js
@@ -1,0 +1,1 @@
+export const isCI = () => process.env.REACT_APP_ENVIRONMENT === 'CI'

--- a/src/common/useRemoteJSON.jsx
+++ b/src/common/useRemoteJSON.jsx
@@ -27,8 +27,10 @@ export function useRemoteJSON(sourceUrl, options = {}) {
 
     fetch(sourceUrl)
       .then((response) => {
-        if (hasHttpError(response)) return Promise.reject(response)
-        return response.json()
+        return hasHttpError(response).then(res => {
+          if(res) return Promise.reject(response)
+          return response.json()
+        })
       })
       .then((json) => {
         if (options.transformReceive) setData(options.transformReceive(json))

--- a/src/tools/rate-spread/CSVUpload.jsx
+++ b/src/tools/rate-spread/CSVUpload.jsx
@@ -2,6 +2,7 @@ import React, { Component } from 'react'
 import fileSaver from 'file-saver'
 import LoadingIcon from '../../common/LoadingIcon.jsx'
 import Alert from '../../common/Alert.jsx'
+import { isCI } from '../../common/environmentChecks'
 import Heading from '../../common/Heading.jsx'
 import runFetch from './runFetch.js'
 
@@ -64,7 +65,7 @@ class CSVUpload extends Component {
     event.target.value = null
 
     this.onCSVFetch()
-    const CSV_URL = '/public/rateSpread/csv'
+    const CSV_URL = (isCI() ? '' : 'https://ffiec.cfpb.gov') + '/public/rateSpread/csv'
     runFetch(CSV_URL, this.prepareCSVBody(file), true).then(res => {
       this.onCSVCalculated(res, file)
     })

--- a/src/tools/rate-spread/Form.jsx
+++ b/src/tools/rate-spread/Form.jsx
@@ -1,6 +1,7 @@
 import React, { Component } from 'react'
 import LoadingIcon from '../../common/LoadingIcon.jsx'
 import Alert from '../../common/Alert.jsx'
+import { isCI } from '../../common/environmentChecks'
 import Heading from '../../common/Heading.jsx'
 import runFetch from './runFetch.js'
 
@@ -175,7 +176,7 @@ class Form extends Component {
       if (errs.rateSetDate || errs.APR || errs.loanTerm) return
 
       this.onFetch()
-      const API_URL = '/public/rateSpread'
+      const API_URL = (isCI() ? '' : 'https://ffiec.cfpb.gov') + '/public/rateSpread'
       runFetch(API_URL, this.prepareBodyFromState()).then(res => {
         this.onCalculated(res)
       })

